File: php/features.php
<?php
declare(strict_types=1);

// CatherineShulmansQuotes SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class CatherineShulmansQuotesFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new CatherineShulmansQuotesBaseFeature();
            case "test":
                return new CatherineShulmansQuotesTestFeature();
            default:
                return new CatherineShulmansQuotesBaseFeature();
        }
    }
}

File: py/features.py
# CatherineShulmansQuotes SDK feature factory

from feature.base_feature import CatherineShulmansQuotesBaseFeature
from feature.test_feature import CatherineShulmansQuotesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CatherineShulmansQuotesBaseFeature(),
        "test": lambda: CatherineShulmansQuotesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
